Week6/Day3/CommunityShedMVC/CommunityShedMVC/Controllers/HomeController.cs

```diff
@@ -0,0 +1,198 @@
+            communities = DatabaseHelper.Retrieve<Community>(@"
+                select Id, CommunityName, [Open], OwnerId 
+                from Community 
+            ");
+            JoinCommunity viewModel = new JoinCommunity();
+            viewModel.Communities = communities;
+
+            return View(viewModel);
+        }
+
+        
+        public ActionResult AddCommunity()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        public ActionResult AddCommunity(Community community)
+        {
+            try
+            {
+                community.OwnerId = CustomUser.Person.Id;
+                //Add new entry for community
+                community.Id = (int)DatabaseHelper.Insert(@"
+                    Insert into Community(CommunityName, [Open], OwnerId)
+                    values (@CommunityName, @Open, @OwnerId)
+
+                ", new SqlParameter("@CommunityName", community.CommunityName),
+                new SqlParameter("@Open", community.Open),
+                new SqlParameter("@OwnerId", community.OwnerId)
+                );
+                //Add new entry for PersonRole
+                DatabaseHelper.Insert(@"
+                    Insert into PersonRole(PersonId, RoleId, CommunityId)
+                    values (@PersonId, @RoleId, @CommunityId)
+
+                ", new SqlParameter("@PersonId", CustomUser.Person.Id),
+                new SqlParameter("@RoleId", 1),
+                new SqlParameter("@CommunityId", community.Id)
+                );
+
+                DatabaseHelper.Insert(@"
+                    Insert into PersonRole(PersonId, RoleId, CommunityId)
+                    values (@PersonId, @RoleId, @CommunityId)
+
+                ", new SqlParameter("@PersonId", CustomUser.Person.Id),
+               new SqlParameter("@RoleId", 2),
+               new SqlParameter("@CommunityId", community.Id)
+               );
+                DatabaseHelper.Insert(@"
+                    Insert into PersonRole(PersonId, RoleId, CommunityId)
+                    values (@PersonId, @RoleId, @CommunityId)
+
+                ", new SqlParameter("@PersonId", CustomUser.Person.Id),
+               new SqlParameter("@RoleId", 3),
+               new SqlParameter("@CommunityId", community.Id)
+               );
+                DatabaseHelper.Insert(@"
+                    Insert into PersonRole(PersonId, RoleId, CommunityId)
+                    values (@PersonId, @RoleId, @CommunityId)
+
+                ", new SqlParameter("@PersonId", CustomUser.Person.Id),
+               new SqlParameter("@RoleId", 4),
+               new SqlParameter("@CommunityId", community.Id)
+               );
+                SetOwnerRoles(community.Id);
+
+                return RedirectToAction("Index", "Home");
+            }
+            catch
+            {
+                return View();
+            }
+
+        }
+
+        
+        public ActionResult JoinCommunity(int? communityId)
```

Comment:
This action method changes data on the server, so be sure to add an `HttpPost` attribute to this method.